internal/config/dash.go

```diff
@@ -94,49 +97,59 @@ type Dash interface {
 	Validate() error
 
 	ModuleManager() module.ManagerInterface
+
+	// ClusterClientManager()
 }
 
 // Live is a live version of dash config.
 type Live struct {
-	clusterClient      cluster.ClientInterface
-	crdWatcher         CRDWatcher
-	logger             log.Logger
-	moduleManager      module.ManagerInterface
-	objectStore        store.Store
-	pluginManager      plugin.ManagerInterface
-	portForwarder      portforward.PortForwarder
-	kubeConfigPath     string
-	currentContextName string
-	restConfigOptions  cluster.RESTConfigOptions
+	clusterClient        cluster.ClientInterface // clusterClientManager
+	clusterClientManager client.ClusterClientManager
+	crdWatcher           CRDWatcher
+	logger               log.Logger
+	moduleManager        module.ManagerInterface
+	objectStore          store.Store
+	pluginManager        plugin.ManagerInterface
+	portForwarder        portforward.PortForwarder
+	kubeConfigPath       string
+	currentContextName   string // remove .. replace with CurrentContext() method ..
+	restConfigOptions    cluster.RESTConfigOptions
 }
 
 var _ Dash = (*Live)(nil)
 
 // NewLiveConfig creates an instance of Live.
 func NewLiveConfig(
-	clusterClient cluster.ClientInterface,
+	clusterClientManager client.ClusterClientManager,
 	crdWatcher CRDWatcher,
 	kubeConfigPath string,
 	logger log.Logger,
 	moduleManager module.ManagerInterface,
 	objectStore store.Store,
 	pluginManager plugin.ManagerInterface,
 	portForwarder portforward.PortForwarder,
-	currentContextName string,
+	currentContextName string, // remove
 	restConfigOptions cluster.RESTConfigOptions,
 ) *Live {
 	l := &Live{
-		clusterClient:      clusterClient,
-		crdWatcher:         crdWatcher,
-		kubeConfigPath:     kubeConfigPath,
-		logger:             logger,
-		moduleManager:      moduleManager,
-		objectStore:        objectStore,
-		pluginManager:      pluginManager,
-		portForwarder:      portForwarder,
-		currentContextName: currentContextName,
-		restConfigOptions:  restConfigOptions,
+		crdWatcher:           crdWatcher,
+		clusterClientManager: clusterClientManager,
+		kubeConfigPath:       kubeConfigPath,
+		logger:               logger,
+		moduleManager:        moduleManager,
+		objectStore:          objectStore,
+		pluginManager:        pluginManager,
+		portForwarder:        portForwarder,
+		currentContextName:   currentContextName, // remove
+		restConfigOptions:    restConfigOptions,
+	}
+	clusterClient, err := l.clusterClientManager.Get(context.Background(), currentContextName)
```

Comment:
This context should come from somewhere else. Don't use a `Background` here.

Reply:
Made it a TODO context for now, will get a proper context plumbed down to it.